DownFall/Models/AttackModel.swift

```diff
@@ -8,78 +8,119 @@
 
 import Foundation
 
+struct AttackSlope: Equatable, Decodable {
+    let over: Int
+    let up: Int
+    
+    static var south: AttackSlope {
+        return AttackSlope(over: 0, up: -1)
+    }
+    
+    static var sideways: [AttackSlope] {
+        return [AttackSlope(over: -1, up: 0), AttackSlope(over: 1, up: 0)]
+    }
+    
+    static var diagonals: [AttackSlope] {
+        return [AttackSlope(over: -1, up: 1),
+                AttackSlope(over: 1, up: 1),
+                AttackSlope(over: -1, up: -1),
+                AttackSlope(over: 1, up: -1)]
+    }
+    
+    static var playerPossibleAttacks: [AttackSlope] {
+        return [
+            AttackSlope(over: -1, up: 0),
+            AttackSlope(over: 0, up: 1),
+            AttackSlope(over: 0, up: -1),
+            AttackSlope(over: 1, up: 0)
+        ]
+    }
+}
+
+
 enum AttackType: String, Decodable {
     case targets
     case areaOfEffect
+    case charges
+    case slope
```

Comment:
I dont think you use slope anymore.  Also, a lot of these monsters should charge.  Consider how to add it as a separate type on attackmodels